velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -316,4 +318,40 @@ struct ArraySumFunction {
   }
 };
 
+/// Function Signature: shuffle(x) -> array
+/// Generate a random permutation of the given array x.
+template <typename TExecCtx, typename T>
+struct ArrayShuffleFunction {
```

Comment:
I feel this function would need to be implemented as a vector function to (1) add support for all types of input including arrays of arrays, maps and structs; (2) avoid copying the input elements.

Reply:
My feeling is the simple API can support nested complex types, will discuss with @laithsakka on this requirement. There are more UDFs like this one.

For scalar types, shall we stick to simple API? And we can add support for complex types down the line.